components/kserve/kserve.go

```diff
@@ -166,6 +183,43 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client, resC
 	return nil
 }
 
+func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	inferenceServiceConfigMap := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{
+		Namespace: dscispec.ApplicationsNamespace,
+		Name:      "inferenceservice-config",
+	}, inferenceServiceConfigMap)
+	if err != nil {
+		fmt.Println("error to get configmap 'inferenceservice-config'")
+		return err
+	}
+
+	//set data.deploy.defaultDeploymentMode to the model specified in the Kserve spec
+	var deployData map[string]interface{}
+	json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData)
+	deployData["defaultDeploymentMode"] = k.DefaultDeploymentMode
+	deployDataBytes, _ := json.MarshalIndent(deployData, "", " ")
```

Comment:
Missing error handling

Reply:
Thanks for spotting it @terrytangyuan. Interestingly enough we have linter gh action which reports that too https://github.com/opendatahub-io/opendatahub-operator/actions/runs/7894716777/job/21545878156?pr=864#step:5:33, though I was expecting to comment back on the PR instead of just dumping it in the logs.

I am lacking enough karma but I think what we are missing is `write` permission for this GH Action https://github.com/golangci/golangci-lint-action#comments-and-annotations

@zdtsw can someone from the team enable it?

Reply:
https://github.com/opendatahub-io/opendatahub-operator/pull/865

Reply:
LOL I think I was lacking enough caffeine at the time of writing it, I thought it was a repo/action setting somewhere like with the gh tokens.

Reply:
guess the only way to know if this PR^ works is a rebase :P

Reply:
> guess the only way to know if this PR^ works is a rebase :P

yeah this is by design, otherwise you could raise malicious PR with elevated privileges and do some weird stuff :)